### PR TITLE
Update higlass

### PR DESCRIPTION
**Pull request by BethYates**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/sanger-tol/genomenote/tree/master/.github/CONTRIBUTING.md)
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by BethYates** [comment]
Update_higlass_server module expects to be passed two files a '.mcool' file and a '.bedpe' file. The actual files that get uploaded are copied to the directory specified by the 'higlass_upload_directory' param by the publishDir directive used by the module that produces each file type. This seems a bit of a hacky way to do it but it works. The publishdir directive for Update_higlass_server.nf means that we still get the expected files in the results/contact_maps directory.

**Comment by muffato** [comment]
I can confirm that it works and I could do a `test_full` run of the pipeline and see the contact map at http://genome-note-higlass.tol-dev.sanger.ac.uk/l/?d=P4JitQ-fQEuXfxrack5XBA

I'm proposing some improvements in #79

File: docs/usage.md [hunk 1]
```diff
@@ -33,6 +33,16 @@ You will also need to set a nextflow secret to store the API key belonging to yo
   nextflow secrets set TOL_API_KEY '[API key]'
 ```
 
+If you wish to run the optional step that writes the .mcool and .genome files produced by the contact_maps subworkflow to a kubenetes hosted higlass server you will need to set the parameter "update_higlass" to true and provide the configuration information for the kubernetes deployment.
+
+```bash
+   --update_higlass 'true'
+   --higlass_upload_directory  '[Path to ingress directory for kubernetes]'
+   --higlass_deployment_name '[Name of kubernetes deployment]'
+   --higlass_namespace '[Name of the namespace used in the kubernetes cluster]'
```
Comment (gq1) [inline docs/usage.md:42]:
```suggestion
   --higlass_deployment_name '[ Name of Higlass Deployment in kubernetes]'
   --higlass_namespace '[Name of the namespace used for Higlass Deployment in Kubernetes]'
```

File: modules/local/update_higlass_server.nf [hunk 2]
```diff
@@ -0,0 +1,37 @@
+process UPDATE_HIGLASS_SERVER {
```
Comment (gq1) [inline modules/local/update_higlass_server.nf:1]: Can we change this process name?

We are not update the server only loading the data. 

Maybe `upload_higlass_data`?

**Comment by gq1** [comment]
> BA

**Comment by gq1** [comment]
> I can confirm that it works and I could do a `test_full` run of the pipeline and see the contact map at http://genome-note-higlass.tol-dev.sanger.ac.uk/l/?d=P4JitQ-fQEuXfxrack5XBA

You generated the view link manually, not by the pipeline? Will is still trying to generate the links automatically

**Comment by muffato** [comment]
> You generated the view link manually, not by the pipeline? Will is still trying to generate the links automatically

Yes, I created the link manually. Creating the link automatically is out of scope for this ticket (TOLIT-427) but it's something we should consider part of the epic (TOLIT-1601)

File: modules/local/update_higlass_server.nf (hunk 2 above)
Comment (BethYates) [reply modules/local/update_higlass_server.nf]: changes made.

File: nextflow_schema.json [hunk 3]
```diff
@@ -78,6 +78,37 @@
...
+                "higlass_kubeconfig": {
+                    "type": "string",
+                    "format": "file-path",
+                    "description": "The path to the kubeconfig  for the kubernetes cluster running the HiGlass server.",
+                    "fa_icon": "fas fa-folder-open",
+                    "hidden": true
+                },
+                "higlass_deployment_name": {
+                    "type": "string",
+                    "description": "The kubernetes cluster deployment name for the HiGlass server.",
```
[the remark this answers is not in the document]
Comment (BethYates) [reply nextflow_schema.json]: I tweaked this slightly to more closely match the text you suggested be used in the usage.md file 
"description": "Name of the kubernetes deployment for the HiGlass server.",

**Review by priyanka-surana — COMMENTED** [review]
I do not have a `kubeconfig` so that step was skipped. I think @gq1 should test this. Otherwise all the steps ran successfully with the `test` profile.

**Comment by gq1** [comment]
> I do not have a `kubeconfig` so that step was skipped. I think @gq1 should test this. Otherwise all the steps ran successfully with the `test` profile.

@muffato has tested all the uploading.

**Pull request merged**